src/optimizer/pullup/pullup_from_left.cpp

```diff
@@ -0,0 +1,29 @@
+#include "duckdb/optimizer/filter_pullup.hpp"
+#include "duckdb/planner/operator/logical_comparison_join.hpp"
+#include "duckdb/planner/operator/logical_join.hpp"
+
+namespace duckdb {
+using namespace std;
+
+unique_ptr<LogicalOperator> FilterPullup::PullupFromLeft(unique_ptr<LogicalOperator> op) {
+	if(op->type == LogicalOperatorType::LOGICAL_COMPARISON_JOIN || op->type == LogicalOperatorType::LOGICAL_ANY_JOIN) {
+		auto &join = (LogicalJoin &)*op;
+		D_ASSERT(join.join_type == JoinType::LEFT || join.join_type == JoinType::SEMI || join.join_type == JoinType::ANTI);
```

Comment:
These assertions seem a bit overkill; perhaps only assert for the operator type?